src/ghostfat.c

```diff
@@ -270,11 +269,11 @@ void uf2_read_block (uint32_t block_no, uint8_t *data)
     for ( uint32_t i = 0; i < FAT_ENTRIES_PER_SECTOR; ++i )
     {
       // Generate the FAT chain for the firmware "file"
-      uint32_t v = (sectionIdx * FAT_ENTRIES_PER_SECTOR) + i;
+      uint32_t v = (sectionIdx * FAT_ENTRIES_PER_SECTOR * BPB_SECTORS_PER_CLUSTER) + (i * BPB_SECTORS_PER_CLUSTER);
 
       if ( UF2_FIRST_SECTOR <= v && v <= UF2_LAST_SECTOR )
```

Comment:
Because `v` is the `fatArrayIndex`, this should use `UF2_FIRST_CLUSTER_NUMBER` and `UF2_LAST_CLUSTER_NUMBER`.  Those are not defined, but recommended in prior comment.